src/csync/csync_exclude.cpp

```diff
@@ -415,8 +438,90 @@ static CSYNC_EXCLUDE_TYPE _csync_excluded_common(c_strlist_t *excludes, const ch
     return match;
 }
 
-CSYNC_EXCLUDE_TYPE csync_excluded_traversal(c_strlist_t *excludes, const char *path, int filetype) {
-  return _csync_excluded_common(excludes, path, filetype, false);
+/* Only for bnames (not paths) */
+static QString convertToBnameRegexpSyntax(QString exclude)
+{
+    QString s = QRegularExpression::escape(exclude).replace("\\*", ".*").replace("?", ".");
+    //qDebug() << "Converted pattern" << exclude << "to regex" << s;
+    return s;
+}
+
+void csync_exclude_traversal_prepare(CSYNC *ctx)
```

Comment:
wondering if this should not be a member function of  the parsed_traversal_excludes class